File: src/Java/101-200/129.TreeSumRootLeafNumbers.java

import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Queue;
import java.util.HashSet;
import java.util.LinkedList;

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {

    private int sum(TreeNode root, int sum) {
        if (root == null) {
            return 0;
        }
        sum = 10 * sum + root.val;
        if (root.left == null && root.right == null) {
            return sum;
        }
        return sum(root.left, sum) + sum(root.right, sum);
    }

    public int sumNumbers(TreeNode root) {
        if (root == null) {
            return 0;
        }
        return sum(root, 0);
    }
}

File: src/Python/101-200/129.TreeSumRootLeafNumbers.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def dfs(self, node, val):
        if node is None:
            return 0
        val = 10 * val + node.val
        if node.left is None and node.right is None:
            return val
        return self.dfs(node.left, val) + self.dfs(node.right, val)

    def sumNumbers(self, root):
        if root is None:
            return 0
        return self.dfs(root, 0)


if __name__ == "__main__":
    solution = Solution()
    node = TreeNode(1)
    node.left = TreeNode(2)
    node.right = TreeNode(3)
    print(solution.sumNumbers(node))
    node.left.left = TreeNode(2)
    node.right.right = TreeNode(3)
    print(solution.sumNumbers(node))
